File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int l=0,r=1;
        int max=0;
        if(prices.length==1)return 0;
        while(r<prices.length){
            if(prices[l]>prices[r]){
                l=r;
                r+=1;
            }else if(prices[l]==prices[r]){
                r+=1;
            }else{
                max=Math.max(max, prices[r]-prices[l]);
                r+=1;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)
        minm = r
        def getHours(rate):
            res = 0
            for pile in piles:
                res += math.ceil(pile/rate)
            return res
        
        while l<=r:
            m = (l+r)//2
            hours = getHours(m)
            if hours > h:
                l = m + 1
            elif hours <= h:
                r = m - 1
                minm = min(minm, m)
            
        return minm
